sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/MapToKeys.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * {@code MapToKeys} maps a {@code SerializableFunction} over keys of a {@code PCollection} of
+ * {@code KV<K1, V>}s and returns a {@code PCollection} of the keys.
+ *
+ * <p>Example of use:
+ *
+ * <pre>{@code
+ * PCollection<KV<Integer, String>> input = ...;
+ * PCollection<Double> output =
+ *      input.apply(MapToKeys.via(Integer::doubleValue));
+ * }</pre>
+ *
+ * <p>See also {@link MapToValues}.
+ *
+ * @param <K1> the type of the keys in the input {@code PCollection}
+ * @param <K2> the type of the elements in the output {@code PCollection}
+ */
+public class MapToKeys<K1, K2>
```

Comment:
I'm not sure about `MapToKeys` and `MapToValues`, this can already be accomplished pretty concisely with:
```
pc.apply(Keys.create())
    .apply(MapElements.into(type).via(func))
```